app/src/main/java/com/anysoftkeyboard/gesturetyping/GestureTypingDetector.java

```diff
@@ -54,10 +57,17 @@
     private final ReplaySubject<LoadingState> mGenerateStateSubject = ReplaySubject.createWithSize(1);
     private final ArrayList<int[]> mWordsCorners = new ArrayList<>();
 
-    public GestureTypingDetector(int curvatureSize, @NonNull Iterable<Keyboard.Key> keys) {
-        mCurvatureSize = curvatureSize;
+    public GestureTypingDetector(int minPointDistance, @NonNull Iterable<Keyboard.Key> keys) {
+        mMinPointDistance = minPointDistance;
         mKeys = keys;
 
+        for (Keyboard.Key key : mKeys) {
```

Comment:
Can this be moved into the `generateCornersInBackground` method? So the calculation would be done in a background thread and not on the UI thread (which calls this constructor)

Reply:
I moved it to a background thread. Please check whether this looks right, I'm pretty unfamiliar with RxJava...